unit_tests/test_zaza_model.py

```diff
@@ -352,6 +352,69 @@ async def _wrapper():
         self.Model_mock.connect_model.assert_called_once_with('modelname')
         self.Model_mock.disconnect.assert_called_once_with()
 
+    def _mocks_for_block_until_auto_reconnect_model(
+            self, is_connected=True, is_open=True):
+        self.patch_object(model, 'Model')
+        self.Model.return_value = self.Model_mock
+        self.Model_mock.is_connected.return_value = is_connected
+        mock_connected = mock.MagicMock()
+        mock_connected.is_open = is_open
+        self.Model_mock.connected.return_value = mock_connected
+        self.patch('asyncio.sleep', name='mock_sleep', new=mock.AsyncMock())
+
+    def _wrapper_block_until_auto_reconnect_model(
+            self, *conditions, aconditions=None, timeout=None, wait_period=0.5,
+            loop=None):
+
+        async def _wrapper():
+            async with model.run_in_model('modelname') as mymodel:
+                await model.block_until_auto_reconnect_model(
+                    *conditions,
+                    aconditions=aconditions,
+                    timeout=timeout,
+                    wait_period=wait_period,
+                    model=mymodel,
+                    loop=loop)
+
+        self._wrapper = _wrapper
+
+    def test_block_until_auto_reconnect_model(self):
+        self._mocks_for_block_until_auto_reconnect_model(True, True)
+        self._wrapper_block_until_auto_reconnect_model(
+            lambda: True)
+        loop.run(self._wrapper())
```

Comment:
Shouldn't we add some assertions here? e.g. that `disconnect()` wasn't called?

Reply:
I mean we can.  It doesn't *really* add anything, but I'll add it for completeness.

Reply:
Actually, it turned out to be extremely useful.  The test code was wrong; see the next iteration.  Thanks for the suggestion.